### PR TITLE
Hide EnvAutodetectResource ro reduce public classes

### PR DESCRIPTION
**Pull request by dengliming**
for [https://github.com/open-telemetry/opentelemetry-java/issues/1472](https://github.com/open-telemetry/opentelemetry-java/issues/1472)  
@carlosalberto @anuraaga Please take a look.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java [hunk 1]
```diff
@@ -117,6 +117,19 @@ public static Resource create(Attributes attributes) {
     return new AutoValue_Resource(attributes);
   }
 
+  /**
+   * Returns a {@link Resource}. This resource information is loaded from the
+   * OTEL_RESOURCE_ATTRIBUTES environment variable or otel.resource.attributes system properties.
+   *
+   * @return a {@code Resource}.
+   */
+  public static Resource createDefault() {
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java:126]: I'd suggest creating this value only once, for simplicity and to keep the invariant `EnvAutodetectResource` has - also, for that case, rename it to `getDefault()` too.

Comment (dengliming) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java]: Thanks. Fixed at lastest commit.

File: sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java [hunk 2]
```diff
@@ -53,6 +53,8 @@
               .setAttribute("telemetry.sdk.language", "java")
               .setAttribute("telemetry.sdk.version", readVersion())
               .build());
+  private static final Resource DEFALUT_AUTODETECT_RESOURCE =
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java:56]: DEFAULT ;)

Comment (dengliming) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java]: done. thx.

**Pull request merged**